src/modules/python/pbs_hooks/PBS_cray_atom.PY

```diff
@@ -176,6 +177,60 @@ class HookHelper(object):
         log_with_caller(pbs.EVENT_DEBUG4, 'path is %s' % path)
         return path
 
+    def is_it_exclusive(job):
+        """
+        check to see if the job requested exclusive, or if the
+        nodes are marked exclusive.  This needs to be passed
+        to ATOM.
+        """
+        place = job.Resource_List["place"]
+        log_with_caller(pbs.EVENT_DEBUG4, "place is %s" % place)
+
+        # See if the node sharing value has exclusive
+        vn = pbs.server().vnode(pbs.get_local_nodename())
+        sharing = vn.sharing
+        log_with_caller(pbs.EVENT_DEBUG4, "The sharing value is %s type %s" %
+                        (str(sharing), str(type(sharing))))
+
+        # Uses the same logic as the scheduler (is_excl())
+        if ((sharing == pbs.ND_FORCE_EXCL) or (sharing == pbs.ND_FORCE_EXCLHOST)):
+            return True
+
+        if (sharing == pbs.ND_IGNORE_EXCL):
+            return False
+
+        excls = re.findall(r'excl', str(place))
+        totalexcls = len(excls)
+        if totalexcls > 0:
+            if "group" in str(place):
+                # A group is in the place statement, is it excl 
+                output = re.search(r'(?<=group=)[^:$]*', str(place))
```

Comment:
I'm not sure what this regex does, what does the input/output look like?

Reply:
The input would look like what can be in a place statement.  For example a place statement could be: -lplace=excl:group=myexcl

But overall this section can be made cleaner.  Let me try to simplify it.